source/common/http/http2/codec_impl.cc

```diff
@@ -572,8 +586,9 @@ int ConnectionImpl::onFrameReceived(const nghttp2_frame* frame) {
   ASSERT(frame->hd.type != NGHTTP2_CONTINUATION);
 
   if (frame->hd.type == NGHTTP2_DATA) {
-    if (!trackInboundFrames(&frame->hd, frame->data.padlen)) {
-      return NGHTTP2_ERR_FLOODED;
+    auto status = trackInboundFrames(&frame->hd, frame->data.padlen);
```

Comment:
I wonder if we should create RETURN_IF_ERROR macros in statusor.h (if they don't already exist) to capture this pattern

Reply:
Yeah, I think this will be a nice readability improvement.

Reply:
I will add this macro as a separate PR and modify this code once it is merged.

Reply:
done